### PR TITLE
DF-2867 BMC New Action Add Incident Work Note

### PR DESCRIPTION
**Pull request by mhofert-r7**
## Proposed Changes

Describe the proposed changes:

  - Add new Action: Add Incident Work Note

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `./run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `./run -T tests/example.json -d -j`
    * All actions/triggers shortcut `./run -T all -d -j` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `./run -R tests/example.json -d -j`
    * All actions/triggers shortcut `./run -R all -d -j` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `./run -A -R all -T all`. For single action validation: `./run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`./run ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

**Comment by mhofert-r7** [comment]
## Assessment
### Run
Autogenerated with:
`./run.sh -R tests/add_incident_work_note.json -j -A`
<details>

```
Logging into URL: http://remd-itsm1902.vuln.lax.rapid7.com:8008
Result of authentication request: ****************************VDg8U
rapid7/BMC Remedy ITSM:1.7.0. Step name: add_incident_work_note

{
  "incident": {
    "values": {
      "Request ID": "INC000000000026|INC000000000026",
      "Submitter": "Remedy Application Service",
      "Submit Date": "2008-11-07T05:14:17.000+0000",
      "Assignee Login ID": "Ian",
      "Last Modified By": "ARAdmin",
      "Last Modified Date": "2019-11-12T16:00:00.000+0000",
      "Status": "Assigned",
      "Status-History": {
        "New": {
          "user": "Action Request Installer Account",
          "timestamp": "2019-06-11T18:54:42.000+0000"
        },
        "Assigned": {
          "user": "ARAdmin",
          "timestamp": "2019-11-12T16:00:00.000+0000"
        },
        "Pending": {
          "user": "Action Request Installer Account",
          "timestamp": "2019-06-11T18:54:42.000+0000"
        }
      },
      "Assignee Groups": "1000000001;",
      "InstanceId": "AG00123F73CF5Eqc4TSQTOQxAgc0QB",
      "Vendor Assignee Groups": "1000000001;",
      "Product Categorization Tier 1": "Software",
      "Product Categorization Tier 2": "Software Application/System",
      "Product Categorization Tier 3": "Database Software",
      "Department": "Customer Service",
      "Site Group": "United States",
      "Region": "Americas",
      "Site": "Headquarters, Building 1.31",
      "SRInstanceID": "NA",
      "Entry ID": "INC000000000026",
      "SRMS Registry Instance ID": "SR0011439CCAD4ec8UQwCkOLAQlQAA",
      "InfrastructureEventType": "None",
      "Description": "User needs local Database System installed.",
      "Company": "Calbro Services",
      "Country": "United States",
      "State Province": "New York",
      "City": "New York",
      "Organization": "Information Technology",
      "Assigned Support Organization": "IT Support",
      "Last Name": "Allbrook",
      "First Name": "Allen",
      "Contact Client Type": "Office-Based Employee",
      "VIP": "No",
      "Contact Sensitivity": "Standard",
      "Street": "1114 Eighth Avenue, 31st Floor",
      "Internet E-mail": "A.Allbrook@calbroservices.com",
      "Phone Number": "1 212 5555454 (11)",
      "Categorization Tier 1": "Request",
      "Categorization Tier 2": "Software",
      "Categorization Tier 3": "Install",
      "Site ID": "STE_SOLN0002846",
      "Assigned Group ID": "SGP000000000010",
      "Person ID": "PPL000000000013",
      "Contact Company": "Calbro Services",
      "Service Type": "User Service Request",
      "Incident Number": "INC_CAL_1000024",
      "Urgency": "4-Low",
      "Impact": "4-Minor/Localized",
      "Priority": "Low",
      "Priority Weight": 0,
      "Reported Source": "Email",
      "Assigned Group": "Frontoffice Support",
      "Assignee": "Ian Plyment",
      "Assigned Support Company": "Calbro Services",
      "Owner Support Organization": "IT Support",
      "Owner Group": "Frontoffice Support",
      "Owner Support Company": "Calbro Services",
      "Owner Group ID": "SGP000000000010",
      "Reported Date": "2008-10-01T04:00:00.000+0000",
      "Responded Date": "2008-10-02T12:00:00.000+0000",
      "Last Acknowledged Date": "2008-10-02T12:00:00.000+0000",
      "Direct Contact Internet E-mail": "A.Allbrook@calbroservices.com",
      "Total Transfers": 2,
      "Estimated Resolution Date": "2008-10-06T21:00:00.000+0000",
      "Required Resolution DateTime": "2008-10-06T21:00:00.000+0000",
      "Direct Contact Company": "Calbro Services",
      "Direct Contact Last Name": "Allbrook",
      "Direct Contact First Name": "Allen",
      "Direct Contact Phone Number": "1 212 555-5454 (11)",
      "Direct Contact Organization": "Information Technology",
      "Direct Contact Department": "Customer Service",
      "Direct Contact Region": "Americas",
      "Direct Contact Site Group": "United States",
      "Direct Contact Site": "Headquarters, Building 1.31",
      "Direct Contact Person ID": "PPL000000000013",
      "Direct Contact Street": "1114 Eighth Avenue, 31st Floor",
      "Direct Contact Country": "United States",
      "Direct Contact State/Province": "New York",
      "Direct Contact City": "New York",
      "Direct Contact Zip/Postal Code": "10036",
      "Direct Contact Time Zone": "(GMT-05:00) Eastern Time (US & Canada)",
      "Direct Contact Site ID": "STE_SOLN0002846",
      "Direct Contact Country Code": "1",
      "Direct Contact Area Code": "212",
      "Direct Contact Local Number": "555-5454",
      "Direct Contact Extension": "11"
    },
    "_links": {
      "self": [
        {
          "href": "http://remd-itsm1902.vuln.lax.rapid7.com:8008/api/arsys/v1/entry/HPD:IncidentInterface/INC000000000026%7CINC000000000026"
        }
      ]
    }
  }
}
```
<summary>
$ cat tests/add_incident_work_note.json | docker run --rm   -i rapid7/bmc_remedy_itsm:1.7.0  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
</summary>
</details>

### UI

Screenshots of the plugin being used in Komand for validating use go below this line.
#### Workflow Builder

#### Job

#### Artifact

File: bmc_remedy_itsm/help.md [hunk 1]
```diff
@@ -395,7 +529,7 @@ Example output:
       "Local Phone": "555-5454",
       "Extension": "66",
       "Street": "1114 Eighth Avenue, 31st Floor",
-      "Internet E-mail": "blah@example.com",
+      "Internet E-mail": "J.Unser@calbroservices.com",
```
Comment (jschipp-r7) [inline bmc_remedy_itsm/help.md]: We're standardizing on `user@example.com` for all e-mail addresses in plugin examples.

Comment (jschipp-r7) [reply bmc_remedy_itsm/help.md]: Update for instances in this file.

File: bmc_remedy_itsm/icon_bmc_remedy_itsm/actions/add_incident_work_note/action.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        try:
+            original_incident_response = requests.get(url, headers=headers)
+        except json.JSONDecodeError as e:
+            raise PluginException(preset=PluginException.Preset.INVALID_JSON,
+                                  data=e)
+
+        handler.error_handling(original_incident_response)
+
+        original_incident = komand.helper.clean(original_incident_response.json())
```
Comment (jschipp-r7) [inline bmc_remedy_itsm/icon_bmc_remedy_itsm/actions/add_incident_work_note/action.py]: Possible JSON decode failure here

File: bmc_remedy_itsm/help.md [hunk 3]
```diff
@@ -6,6 +6,140 @@
 
 ## Actions
 
+### Add Incident Work Note
+
+This action is used to add a work note to an incident.
```
Comment (mrinehart-r7) [inline bmc_remedy_itsm/help.md]:
```suggestion
This action is used to add a work note to an Incident.
```

File: bmc_remedy_itsm/plugin.spec.yaml [hunk 4]
```diff
@@ -996,3 +991,23 @@ actions:
         type: incident
         description: "Incident"
         required: false
+  add_incident_work_note:
+    title: Add Incident Work Note
+    description: Add a work note to an incident
```
Comment (mrinehart-r7) [inline bmc_remedy_itsm/plugin.spec.yaml]:
```suggestion
    description: Add a work note to an Incident
```

File: bmc_remedy_itsm/plugin.spec.yaml [hunk 5]
```diff
@@ -1,14 +1,9 @@
...
-product: ["insightconnect"]
 name: bmc_remedy_itsm
 title: "BMC Remedy ITSM"
 description: "Transform your service desk with the right ITSM solution"
-version: 1.6.0
+version: 1.7.0
 vendor: rapid7
-resources:
-  source_url: https://github.com/rapid7/insightconnect-plugins/
-  license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
```
Comment (mrinehart-r7) [inline bmc_remedy_itsm/plugin.spec.yaml]: These should be added back

File: bmc_remedy_itsm/plugin.spec.yaml [hunk 6]
```diff
@@ -1,14 +1,9 @@
 plugin_spec_version: v2
-extension: plugin
-product: ["insightconnect"]
```
Comment (mrinehart-r7) [inline bmc_remedy_itsm/plugin.spec.yaml]: These should be added back

**Pull request merged**